### PR TITLE
Temporary shortcuts + Flip notify, honor change when leaving play, reduce cost prompt 

### PR DESCRIPTION
**Pull request by Kertanos**

### CODE REVIEW

File: o8g/scripts/actions.py [hunk 1]
```diff
@@ -337,6 +337,7 @@ def flip(card, x=0, y=0):
     card.alternate = ALTERNATE_POLITICAL if not card.alternate else ''
     return
   card.isFaceUp = not card.isFaceUp
+  notify("{} flips {}.".format(me,card))
```
Comment (blakewedwards) [inline o8g/scripts/actions.py]: Realize what the problem is seeing the function, the card.isFaceUp toggle will have the generic octgn notify so it's ok, but the if check outputs nothing. Could you move the notify to just above the return and phrase it like ```notify("{} turns '{}' to {}".format(me, card, card.alternate if card.alternate else 'Military'))```

Comment (blakewedwards) [reply o8g/scripts/actions.py]: Would also be good to add a mute() to just the if section in case octgn adds messages for those commands

File: o8g/scripts/actions.py [hunk 2]
```diff
@@ -362,6 +363,12 @@ def has_deck(card, x=0, y=0):
 def discard(card, x=0, y=0):
   pile = get_discard_pile(card)
   if pile is not None:
+    if card.markers[HONORED]:
+      me.honor+=1
+      notify("{} gains 1 honor from {} honored status.".format(me, card))
```
Comment (blakewedwards) [inline o8g/scripts/actions.py]: I haven't muted this function so the notify isn't needed at the moment. In general I've tried to use the generic messages since they describe the actions in the way you'd describe a player normally interacting with the actual cards. What you'll see here if the notify is removed is ```Intolerance sets Intolerance's Honor counter to 12 (+1)
Intolerance moves 'Doomed Shugenja' to Intolerance's Dynasty Discard``` which is the correct ordering due to the honor change being an interrupt

File: o8g/definition.xml [hunk 3]
```diff
@@ -65,27 +65,27 @@
...
     backgroundStyle="uniformToFill"
     ordered="False">
-    <groupaction menu="Set up" execute="setup" showIf="setup_required"/>
-    <groupaction menu="Pass" execute="pass_action" showIf="setup_not_required"/>
-    <groupaction menu="Give honor" execute="give_honor" showIf="setup_not_required"/>
-    <groupaction menu="Declare conflict" execute="declare_conflict" showIf="setup_not_required"/>
-    <groupaction menu="Claim ring" execute="claim_ring" showIf="setup_not_required"/>
-    <groupaction menu="End turn" execute="end_turn" showIf="setup_not_required"/>
-    <groupaction menu="Flip coin" execute="flip_coin" showIf="setup_not_required"/>
+    <groupaction menu="Set up" execute="setup" showIf="setup_required" shortcut="CTRL+SHIFT+S" />
```
Comment (blakewedwards) [inline o8g/definition.xml]: Looks good, I didn't realize that shortcuts got around showIf.. too bad but good to know. I'm a bit curious how you chose some of the shortcuts since they don't match the letters in the word. Does it match the conquest versions or the letters for the word in a different language?

File: o8g/scripts/actions.py [hunk 4]
```diff
@@ -395,12 +402,24 @@ def random_discard_from(group):
 def can_play(card, x=0, y=0):
   return unpack(card, lambda c: c.isFaceUp and (c.type == TYPE_CHARACTER or c.type == TYPE_EVENT or c.type == TYPE_ATTACHMENT))
 
+def prompt_reduce_cost(cost):
```
Comment (blakewedwards) [inline o8g/scripts/actions.py:457]: Looks great

File: o8g/definition.xml (hunk 3 above)
Comment (Kertanos) [reply o8g/definition.xml]: well there is some actions that share letters, so i had to find something else.
Some are also coming from conquest yeah, like ALT+D for pass, was the shortcuts for "i'm done"
Like i said, it's temporary, we can find a better list of shortcuts i think ^^

File: o8g/scripts/actions.py (hunk 2 above)
Comment (Kertanos) [reply o8g/scripts/actions.py]: I wanted to give a bit more information about where the gain or loss of honor was coming from.
Doesn't really matter when you discard only one card, but when you select and discard many, like you would in fate phase, it can be usefull.
But maybe it's too much yeah

File: o8g/scripts/actions.py (hunk 1 above)
Comment (Kertanos) [reply o8g/scripts/actions.py]: Yeah i misunderstood your trello card, thought the isFaceUp function didn't show anything ^^
i'll correct that

**Comment by Kertanos** [comment]
There is a problem with flipping rings and imperial favor.
It can't seems to find the card properties once flipped to notify.
it's also strangely retroactive.
Meaning when i try to notify card or card.name, it shows nothing.
Worse, the name then shows on the previous message when i flip it back.
Then disappear again when you reflip it...

So for now i decided to cancel any changes, and not notify anything yet.
I'll investigate.

File: o8g/Sets/Core Set/set.xml [hunk 5]
```diff
@@ -130,11 +130,11 @@
       <property name="Type" value="Stronghold"/>
       <property name="Unique" value="Unique"/>
       <property name="Traits" value="City."/>
-      <property name="Text" value="???"/>
```
Comment (blakewedwards) [inline o8g/Sets/Core Set/set.xml:133]: Hmm I'm not sure why these changes are showing up, they should already exist and don't need to be merged

File: o8g/Sets/Core Set/set.xml [hunk 6]
```diff
@@ -2327,69 +2472,79 @@
       <property name="Type" value="Role"/>
       <property name="Traits" value="Keeper. Air."/>
       <property name="Text" value="Increase your deckbuilding influence value by 3.&#10;Reaction: After you win an [Air] conflict as the defending player - gain 1 fate."/>
+      <property name="Ring" value="Air"/>
```
Comment (blakewedwards) [inline o8g/Sets/Core Set/set.xml:2475]: The set data is automatically generated so anything manually put here will get overwritten. I think it might be simpler to generate the proxy based on another property of the role until we get better images, like name, traits, or card number

File: o8g/definition.xml [hunk 7]
```diff
@@ -64,32 +64,33 @@
...
-    ordered="False">
-    <groupaction menu="Set up" execute="setup" showIf="setup_required"/>
-    <groupaction menu="Pass" execute="pass_action" showIf="setup_not_required"/>
-    <groupaction menu="Give honor" execute="give_honor" showIf="setup_not_required"/>
-    <groupaction menu="Declare conflict" execute="declare_conflict" showIf="setup_not_required"/>
-    <groupaction menu="Claim ring" execute="claim_ring" showIf="setup_not_required"/>
-    <groupaction menu="End turn" execute="end_turn" showIf="setup_not_required"/>
-    <groupaction menu="Flip coin" execute="flip_coin" showIf="setup_not_required"/>
+    ordered="False"
+    visibility="undefined">
```
Comment (blakewedwards) [inline o8g/definition.xml:68]: Nice

File: o8g/definition.xml [hunk 8]
```diff
@@ -64,32 +64,33 @@
...
+    <cardaction menu="Break/unbreak" execute="toggle_break" showIf="is_province" shortcut="CTRL+B" />
+    <cardaction menu="Discard" execute="discard" showIf="has_deck" shortcut="DEL" />
+    <cardaction menu="Flip" execute="flip" showIf="can_flip" shortcut="F8" />
+    <cardaction menu="Honor" execute="honor" showIf="can_honor" shortcut="F3" />
+    <cardaction menu="Dishonor" execute="dishonor" showIf="can_honor" shortcut="F4" />
+    <cardaction menu="Play" execute="play_dynasty" showIf="can_play" shortcut="F7" />
+    <cardaction menu="Refill" execute="refill" showIf="is_province" shortcut="CTRL+R" />
+    <cardaction menu="Replace" execute="replace" showIf="can_replace" shortcut="CTRL+T" />
     <cardaction menu="Select bid" execute="select_bid" showIf="is_honor_dial"/>
-    <cardaction menu="Move to bottom of deck" execute="move_to_deck_bottom" showIf="has_deck"/>
```
Comment (blakewedwards) [inline o8g/definition.xml:89]: I didn't realize this already existed in the generic move-to menu.

**Comment by blakewedwards** [comment]
Looks great, only things I'm not sure about are why some of the set diffs are appearing in the PR and the new fields in the set data.

File: o8g/Sets/Core Set/set.xml (hunk 6 above)
Comment (Kertanos) [reply o8g/Sets/Core Set/set.xml:2475]: Ok i'll try something like that

File: o8g/definition.xml (hunk 8 above)
Comment (Kertanos) [reply o8g/definition.xml:89]: and you can also just drag and drop the cards to the bottom of the deck (there is a little arrow that shows you you're moving to bottom instead of top)

**Comment by Kertanos** [comment]
There you go, Role elemental rings now depends from Traits instead or Rings.
Not sure what i can do about the weird set.xml "changes that are not changes because it's already in both version"
